src/signal.rs

```diff
@@ -1796,3 +1812,31 @@ mod status_icon {
         f(&from_glib_none(this), size).to_glib()
     }
 }
+
+pub trait ScaleSignals {
+    fn connect_format_value<F: Fn(&Self, f64) -> String + 'static>(&self, f: F) -> u64;
+}
+
+mod scale {
+    use Scale;
+    use libc::{c_double, c_char};
+    use std::mem::transmute;
+    use ffi::GtkScale;
+    use glib::signal::connect;
+    use glib::translate::*;
+
+    impl super::ScaleSignals for Scale {
+        fn connect_format_value<F: Fn(&Self, f64) -> String + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<F>> = Box::new(Box::new(f));
```

Comment:
The trouble is in this line: the inner box needs to be a trait object `Box<Fn(&Scale, f64) -> String + 'static>`, `Box<F>` is just a boxed struct.